v1/contracts/Store.sol

```diff
@@ -0,0 +1,60 @@
+/*
+  Store implementation.
+ 
+  An implementation of StoreInterface with a fee per second and withdraw functions for the owner.
+*/
+pragma solidity ^0.5.0;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+//import "./StoreInterface.sol";
+import "./Withdrawable.sol";
+
+
+// An implementation of StoreInterface that can accept Oracle fees in ETH or any arbitrary ERC20 token.
+contract Store {//is Withdrawable {
+
+    using SafeMath for uint;
+
+    uint private fixedOracleFeePerSecond; // Percentage of 10^18. E.g., 1e18 is 100% Oracle fee.
+    uint private constant FP_SCALING_FACTOR = 10**18;
+
+    uint weeklyDelayFee;
+    mapping(address => uint) finalFees;
+    uint secondsPerWeek = 604800;
+
+    function payOracleFees() external payable {
+        require(msg.value > 0);
+    }
+
+    function payOracleFeesErc20(address erc20Address) external {
+        IERC20 erc20 = IERC20(erc20Address);
+        uint authorizedAmount = erc20.allowance(msg.sender, address(this));
+        require(authorizedAmount > 0);
+        require(erc20.transferFrom(msg.sender, address(this), authorizedAmount));
+    }
+
+    // Sets a new Oracle fee per second.
+    //TODO change permissioning
+    function setFixedOracleFeePerSecond(uint newOracleFee) external {
+        // Oracle fees at or over 100% don't make sense.
+        require(newOracleFee < FP_SCALING_FACTOR);
+        fixedOracleFeePerSecond = newOracleFee;
+        emit SetFixedOracleFeePerSecond(newOracleFee);
+    }
+
+    function computeRegularFee(uint startTime, uint endTime, uint pfc, bytes32 identifier) external view returns (uint regularFee, uint latePenalty) {
```

Comment:
We have a new FixedPoint library: https://github.com/UMAprotocol/protocol/blob/3c77112fe7635a4519775ec4f52a73eccc4d581c/v1/contracts/FixedPoint.sol#L17. So I would take `pfc` as a FixedPoint.Unsigned and return `regularFee` and `latePenalty` as FixedPoint.Unsigned numbers.

To use this, you'll probably need to add `using FixedPoint for FixedPoint.Unsigned;` and `using FixedPoint for uint;` at the top of this contract. You'll also need to import the `FixedPoint.sol` file.